test/commands/commerce/extension/unmap.test.ts

```diff
@@ -0,0 +1,92 @@
+    // const sfdxError = new SfdxError('error');
+
+    after(() => {
+        sinon.restore();
+    });
+    it('Throws error with a invalid Webstore Name', async () => {
+        const forceDataSoqlStub = sinon.stub(forceOrgSoqlExports, 'forceDataSoql');
+        forceDataSoqlStub.withArgs(QUERY_GET_WEBSTORE, orgUserName).throws(new TypeError('Invalid Webstore'));
+        assert.throws(
+            () => unmapCommand.unmapRecord(registeredExtensionName, storeName, undefined, orgUserName),
+            TypeError
+        );
+        forceDataSoqlStub.restore();
+    });
+    it('Throws error with a invalid Webstore Id', async () => {
+        const forceDataSoqlStub = sinon.stub(forceOrgSoqlExports, 'forceDataSoql');
+        forceDataSoqlStub.withArgs(QUERY_GET_WEBSTORE_ID, orgUserName).throws(new SfdxError('Invalid Store Id'));
+        assert.throws(
+            () => unmapCommand.unmapRecord(registeredExtensionName, storeName, storeId, orgUserName),
+            SfdxError
+        );
+        forceDataSoqlStub.restore();
+    });
+    it('Throws error with deleting a record', async () => {
+        const forceDataSoqlStub = sinon.stub(forceOrgSoqlExports, 'forceDataSoql');
+        const qr = new Result<QueryResult>();
+        qr.result = new (class implements QueryResult {
+            public done: boolean;
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            public records: Record[] = [{ Id: 'hi' }];
+            public totalSize = 1;
+        })();
+        forceDataSoqlStub.withArgs(QUERY_GET_WEBSTORE_ID, 'testUserName').returns(qr);
+        // stub EPN query call with size 1 to let it flow through the code
+        const epnQr = new Result<QueryResult>();
+        epnQr.result = new (class implements QueryResult {
+            public done: boolean;
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            public records: Record[] = [{ Value: 'bye' }];
+            public totalSize = 1;
+        })();
+        forceDataSoqlStub.withArgs(QUERY_GET_STORENAME, 'testUserName').returns(epnQr);
+        // // stub insert record call
+        const recordQr = new Result<QueryResult>();
+        recordQr.result = new (class implements QueryResult {
+            public done: boolean;
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            public records: Record[] = [];
+            public totalSize = 1;
+        })();
+        const forceDataRecordStub = sinon.stub(forceOrgSoqlExports, 'forceDataRecordDelete');
+        forceDataRecordStub.withArgs(service, DELETE_RECORD, 'testUserName');
+        assert.throws(
+            () => unmapCommand.unmapRecord(registeredExtensionName, storeName, storeId, orgUserName),
+            TypeError
+        );
+        assert(forceDataSoqlStub.calledWith(QUERY_GET_WEBSTORE_ID, orgUserName));
+        forceDataSoqlStub.restore();
```

Comment:
You can call this in after() too, I see this as the last line in every test. Everything else looks good 👌